numpyro/util.py

```diff
@@ -408,3 +408,108 @@ def soft_vmap(fn, xs, batch_ndims=1, chunk_size=None):
         lambda y: jnp.reshape(y, (-1,) + jnp.shape(y)[map_ndims:])[:batch_size], ys
     )
     return tree_map(lambda y: jnp.reshape(y, batch_shape + jnp.shape(y)[1:]), ys)
+
+
+def format_trace_shapes(trace, title="Trace Shapes:", last_site=None, log_prob=False):
```

Comment:
nit: I think `format_shapes` is fine here. Could you add `*` after `trace` so that we can have clearer user codes?